auth0/src/main/java/com/auth0/android/request/internal/GsonProvider.kt

```diff
@@ -0,0 +1,40 @@
+package com.auth0.android.request.internal
+
+import androidx.annotation.VisibleForTesting
+import com.auth0.android.result.Credentials
+import com.auth0.android.result.UserProfile
+import com.auth0.android.util.JsonRequiredTypeAdapterFactory
+import com.google.gson.Gson
+import com.google.gson.GsonBuilder
+import com.google.gson.reflect.TypeToken
+import java.security.PublicKey
+import java.text.SimpleDateFormat
+import java.util.*
+
+internal object GsonProvider {
+    internal val gson: Gson
+    private var sdf: SimpleDateFormat
+    private const val DATE_FORMAT = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
```

Comment:
I thinks this date format could cause issues, as the milliseconds are optional and not always sent by Auth0. We had an issue in `auth0-java` related to this, fixed with https://github.com/auth0/auth0-java/pull/295

Reply:
The library uses this for the user profile deserialization (users API / get profile) and for storing/recovering credentials from the local storage (we first save the credentials with this format and then recover back, it would not impact this case). 

We are using Gson rather than Jackson so the proposed solution of not specifying any date format and defaulting to the library's format could not apply. I'd like to check if the dates that the server sends in the User Profile object are correctly deserialized.

Reply:
It's something to keep in mind in case this does come up - an edge case for sure, but it is possible. Since this isn't new to this PR, we can track it separately and proceed with this PR.